moveit_core/robot_trajectory/src/robot_trajectory.cpp

```diff
@@ -228,24 +232,27 @@ void RobotTrajectory::getRobotTrajectoryMsg(moveit_msgs::msg::RobotTrajectory& t
   if (!onedof.empty())
   {
     trajectory.joint_trajectory.header.frame_id = robot_model_->getModelFrame();
-    trajectory.joint_trajectory.header.stamp = ros::Time(0);
+    trajectory.joint_trajectory.header.stamp = stamp;
     trajectory.joint_trajectory.points.resize(waypoints_.size());
   }
 
   if (!mdof.empty())
   {
     trajectory.multi_dof_joint_trajectory.header.frame_id = robot_model_->getModelFrame();
-    trajectory.multi_dof_joint_trajectory.header.stamp = ros::Time(0);
+    trajectory.multi_dof_joint_trajectory.header.stamp = stamp;
     trajectory.multi_dof_joint_trajectory.points.resize(waypoints_.size());
   }
 
-  static const ros::Duration ZERO_DURATION(0.0);
+  static const rclcpp::Duration ZERO_DURATION(0.0);
   double total_time = 0.0;
+  rclcpp::Duration dur_total(0, 0);
   for (std::size_t i = 0; i < waypoints_.size(); ++i)
   {
     if (duration_from_previous_.size() > i)
       total_time += duration_from_previous_[i];
 
+    dur_total = rclcpp::Duration(static_cast<int64_t>(total_time*1e+9));
```

Comment:
I meant something like this:
```suggestion
    dur_total = rclcpp::Duration(1, 0) * total_time;
```
 I still think it's odd that `rclcpp::Time` and `rclcpp::Duration` don't have double constructors but to me this version would be more readable.

Reply:
Commited! ;)